File: 프로그래머스/1/42862. 체육복/체육복.java
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        
        int[] student = new int[n + 1];
        
        for(int i = 0; i < n; i++){
            if(lost.length - 1 >= i) student[lost[i]]--;
            if(reserve.length - 1 >= i) student[reserve[i]]++;

            if(lost.length - 1 < i && reserve.length - 1 < i) break;
        }
        
        for(int j = 1; j <= n; j++){
            
            // lost에 포함 안되는 학생
            if(student[j] >= 0){
                answer++;
                continue;
            }
            
            // student 요소 - 1 이전이 reserve에 해당 될 경우
            if(j > 0 && student[j - 1] > 0){
                student[j - 1]--;
                student[j]++;
                answer++;
                continue;
            }
            
            // student 요소 + 1 이후가 reserve에 해당 될 경우
            if(j < n && student[j + 1] > 0){
                student[j + 1]--;
                student[j]++;
                answer++;
                continue;
            }
            
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = n
    student = [1] * (n + 1)
    
    for r in reserve:
        student[r] += 1
        
    for l in lost:
        student[l] -= 1
    
    for l in sorted(lost):
        if student[l] == 0:
            if l - 1 >= 0 and student[l - 1] == 2:
                student[l - 1] = 1
            elif l + 1 <= n and student[l + 1] == 2:
                student[l + 1] = 1
            else:
                answer -= 1

    return answer